packages/enzyme-test-suite/test/selector-spec.jsx

```diff
@@ -0,0 +1,324 @@
+import './_helpers/setupAdapters';
+import React from 'react';
+import { expect } from 'chai';
+
+import { describeWithDOM } from './_helpers';
+import {
+  mount,
+  shallow,
+} from 'enzyme';
+
+const tests = [
+  {
+    name: 'mount',
+    renderMethod: mount,
+    describeMethod: describeWithDOM,
+  },
+  {
+    name: 'shallow',
+    renderMethod: shallow,
+    describeMethod: describe,
+  },
+];
+
+describe('selectors', () => {
+  tests.forEach(({ describeMethod, name, renderMethod }) => {
+    describeMethod(name, () => {
+      it('simple descendent', () => {
+        const wrapper = renderMethod(
+          <div>
+            <div className="top-div">
+              <span>inside top div</span>
+            </div>
+
+            <div className="bottom-div" />
+            <span />
+          </div>,
+        );
+
+        expect(wrapper.find('span')).to.have.lengthOf(2);
+        expect(wrapper.find('.top-div span')).to.have.lengthOf(1);
+      });
+
+      it('nested descendent', () => {
+        const wrapper = renderMethod(
+          <div>
+            <div className="my-div">
+              <h1>heading</h1>
+              <div>
+                <div className="my-div">
+                  <h1>heading</h1>
+                </div>
+              </div>
+            </div>
+            <h1>heading</h1>
+          </div>,
+        );
+
+        expect(wrapper.find('h1')).to.have.lengthOf(3);
+        expect(wrapper.find('.my-div h1')).to.have.lengthOf(2);
+      });
+
+      it('deep descendent', () => {
+        const wrapper = renderMethod(
+          <div>
+            <div>
+              <div className="inner">
+                <span>
+                  <div className="way-inner">
+                    <h1>heading</h1>
+                  </div>
+                </span>
+              </div>
+            </div>
+            <h1>heading</h1>
+          </div>,
+        );
+
+        expect(wrapper.find('h1')).to.have.lengthOf(2);
+        expect(wrapper.find('div .inner span .way-inner h1')).to.have.lengthOf(1);
+      });
+
+      it('direct descendent', () => {
+        const wrapper = renderMethod(
+          <div>
+            <div className="container">
+              <div className="to-find">Direct</div>
+              <div>
+                <div className="to-find">Nested</div>
+              </div>
+            </div>
+            <div className="to-find">Outside</div>
+          </div>,
+        );
+
+        expect(wrapper.find('.to-find')).to.have.lengthOf(3);
+        const descendent = wrapper.find('.container > .to-find');
+        expect(descendent).to.have.lengthOf(1);
+        expect(descendent.text()).to.equal('Direct');
+      });
+
+      it('simple adjacent', () => {
+        const wrapper = renderMethod(
+          <div>
+            <div className="to-find" />
```

Comment:
something worth testing is to have children of a node that are in separate arrays but adjacent to one another:

```
<div>
  <div className="a" />
  {[<div key="0" className="b" />]}
</div>
```
With this you can assert:
```
expect(wrapper.find('.a + .b'))..to.have.lengthOf(1)
```